### ISSUE TITLE
KomodoIDE 10 beta: CompareTool is launched permanently

### ISSUE THREAD

**Issue by hoppfrosch** [Komodo/KomodoEdit#1433]
### Short Summary
Each time starting Komodo 10, my Compare-Tool is launched (opening a random comparison of a file from project against its VCS-version). Closing the CompareTool and reactivating Komodo relaunches the CompareTool (files/folders to be compared look randomly choosen)

In my production environment I have configured BeyondCompare as external CompareTool for Komodo and everything works as expected. Installing KomodoIde 10 beta the settings for external CompareTool are taken automatically from KomodoIDE9. I haven't found an preference setting yet in Komodo10, to change the external CompareTool 

### Steps to Reproduce
?

### Expected results
Starting and reactivating Komodo should not launch my external CompareTool

### Actual results
see above

### Platform Information
KomodoIde beta 10  build 89113 (and previous)
Win7 Ultimate SP1
git 2.8.0 for windows as VCS
BeyondCompare 3.3.13

**Comment by Naatan** [comment]
Could you please explain how you configured your system to use a specific compare tool?

**Comment by hoppfrosch** [comment]
Hmmmm ... I did this quite a while ago at Release of Komodo 8 and cannot remember exactly. Yet I cannot find any options anymore to configure the difftool exactly. Within my git (which I use as VCS) I've cinfigured BeyondCompare as difftool with the following settings:

diff.tool=bc3
diff.guitool=bc3
difftool.prompt=false
difftool.bc3.cmd='D:/Portable/PortableApps/BeyondCompare/BComp.exe' "$LOCAL" "$REMOTE"
difftool.sourcetree.cmd='D:/Portable/PortableApps/BeyondCompare/BComp.exe' "$LOCAL" "$REMOTE"

What I can say for sure: I have not changed any configuration of my preferred difftool for git since a  loooong time - and the difftool used to work within komodo without any flaws until installing Komodo 10 beta (and it currently still works within Komodo 9 without any errors ...)

**Comment by Naatan** [comment]
I've tried reproducing the `diff.tool` settings (the `difftool.*` settings don't take) but I still cannot reproduce the issue.

If you check Preferences > Environment, are there any variables there that relate to diff tools or git? Can you share them?

**Comment by hoppfrosch** [comment]
Concerning git/difftool: All I found within Preferences > Environment was the following
![komodo_2016-05-12_10-17-33](https://cloud.githubusercontent.com/assets/419075/15208294/0f1295e6-182b-11e6-9fea-656934574da5.png)

**Comment by Naatan** [comment]
I still cannot reproduce it after replicating those settings. Maybe it's a windows only issue though. @cgchoffman could you try?

**Comment by cgchoffman** [comment]
I'm not able to reproduce this either.  Komodo starts normally and no compare appears with random files.  This sounds like a Userscript or Addon is running at start up. @hoppfrosch, have you tried Help > Troubleshooting > Restart without Addons / Restart without tools?

**Comment by hoppfrosch** [comment]
@cgchoffman: Just tested it both variants (wo AddOns/ wo Tools) - with the same effect: the DiffTool appears .... :-(

Where do I configure the difftool to be used with Komodo? I cannot remember how I did this a long time ago ...

**Comment by Naatan** [comment]
Komodo doesn't currently offer a way to specify your own diff app.

Could you try removing the GIT_EXTERNAL_DIFF env variable? Although using that did not give me the same issue as you, it did give me some odd behaviour.

**Comment by hoppfrosch** [comment]
I'll try removing ...

> Komodo doesn't currently offer a way to specify your own diff app.

What are the checkboxes for external diff within komodo preferences for?

**Comment by hoppfrosch** [comment]
I failed removing GIT_EXTERNAL_DIFF via Komodo preferences since it's set as "Startup environment variable" and does not activate the deltre-button when selecting it from the environment listing ...

**Comment by Naatan** [comment]
Try adding it to the user env variables and making the value empty

**Comment by hoppfrosch** [comment]
I did ... and everything works like expected - but my external difftool is not available anymore ;-)

I think, I found a possible reason:
The button pane has a few tabs ("Version Control", "Breakpoints" ....). Starting Komodo a certain console is opened - in my case the "first" (leftmost) tab "Version Control" - with the sub-Tab "Commit Changes". The "Commit Changes" contain a diff view ....

Each time I open Komodo the "Version Control"-"Commit changes"-Tab on the bottom panel is opened and the changes of the last opened file are to be shown. Having configured an external-difftool this launches the external difftool

I tried to reorder the tabs on the bottom panel - and restarted after reording. But the original ordering was restored after this (having the "Version Control"-"Commit changes"-Tab  as a first tab).

**Comment by hoppfrosch** [comment]
I filed a new issue: #1519 concerning persistent storage of pane tab ordering

**Comment by Naatan** [comment]
That is indeed what is triggering a diff, but that is not the cause of the issue.

So forcing GIT_EXTERNAL_DIFF to be empty solved the issue, but now you cannot trigger your own external diff anymore (I assume you want this to show when explicitly opening a diff from within Komodo), do I have that right?

**Comment by hoppfrosch** [comment]
Yes - explicitly opening a diff from within komodo does not work without having GIT_EXTERNAL_DIFF set (as expected)

**Comment by Naatan** [comment]
Alright, perfect. I understand the problem and your desired use-case now. Thank you!

**Comment by hoppfrosch** [comment]
Just as a hint: git_GUIs like Atlassian SourceTree do have an external diff. This is only opened, if the user triggers is, when the Application itself needs to display it's own diff view. Applied to Komodo in my use case would look like this: if  "Version Control"-"Commit changes"-Tab is opened, Komodo does not open the external diff, but its own diff-view. If the user wants to show the diffs in external diff, he has to trigger this explicitly (for example via context menu) ...

**Comment by garak** [comment]
Getting same problem with:
Komodo IDE, version 10.0.0, build 89159
Ubuntu 16.04 
git 2.7.4
meld

**Comment by Naatan** [comment]
@hoppfrosch right, that's exactly what I mean by implicit vs explicit :)

**Comment by hoppfrosch** [comment]
@Naatan: As you've might noticed I'm not a native english speaker, so I might miss some details ;-)

**Comment by sfsccn** [comment]
Not sure if the resolution implied above addresses the issues I'm having. I just upgraded from 9.3.0 to 10.0.1. Now when I edit a git-tracked document my graphical diff program (p4merge) pops up whenever I save the document. This is, of course, very disruptive.

My temporary solution is to disable the external git diff tool by removing the [merge], [mergetool] and [diff] sections from ~/.gitconfig.

I'd like to reinstate my external graphical diff too, will the resolution above do this?

I also tried modifying Preferences/SCC/Git by checking "Uses an external diff tool" (I'm really not sure what this does) and adding "--no-ext-diff" to "Diff options" but these had no effect. I didn't find anything relevant in Preferences/Environment.

The problem seems occur no matter what tab is selected in the bottom pane: Version Control, Breakpoints, etc. Not sure if I understood the discussion above on that.

OS: Linux 3.10.0-327.28.2.el7.x86_64
Git: 2.6.1

Help appreciated, thanks!

**Comment by tomwitt2** [comment]
Just another data point. 

This is not a git specific issue.  Komodo causes my subversion external-diff tool to launch multiple times on every file save.  The only place I've defined my diff tool is in my ~/.subversion/config file.  I can only get this crazy behavior to stop by commenting out the diff tool definition in ~/.subversion/config

```
### Section for configuring external helper applications.
[helpers]
### Set diff-cmd to the absolute path of your 'diff' program.
###   This will override the compile-time default, which is to use
###   Subversion's internal diff implementation.
# diff-cmd = diff_program (diff, gdiff, etc.)
# diff-cmd = /home/tom/bin/meld_svn.sh # <--- disabled to stop rampant Komodo diff'ing.
```

**Comment by dooleyk** [comment]
FYI...

I had the same problem with my Subversion diff-cmd in my configuration file on my CentOS7 box, using...
Komodo IDE, version 11.0.2, build 90813, platform linux-x86_64.
Built on Mon Dec 4 10:20:45 2017.
